src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jax_rs/ResourceMethodMultipleEntityParamsQuickFix.java

```diff
@@ -35,54 +44,96 @@
+        entityParamIndexes.forEach(entityParamIndex -> {
+            addCreateCodeAction(context, diagnostic, codeActions, entityParamIndexes, entityParamIndex);
+        });
+        return codeActions;
+    }
+
+    private void addCreateCodeAction(JavaCodeActionContext context, Diagnostic diagnostic, List<CodeAction> codeActions, List<Integer> entityParamIndexes, Integer entityParamIndex) {
+        final JavaCodeActionContext targetContext = context.copy();
+        final PsiElement targetNode = targetContext.getCoveredNode();
+        final PsiParameter[] parameters = PsiTreeUtil.getParentOfType(targetNode, PsiMethod.class).getParameterList().getParameters();
+
+        final String title = getTitle(parameters[entityParamIndex]);
 
-            entityParamIndexes.forEach(entityParamIndex -> {
-                final JavaCodeActionContext targetContext = context.copy();
-                final PsiElement targetNode = targetContext.getCoveredNode();
-                final PsiClass parentType = PsiTreeUtil.getParentOfType(targetNode, PsiClass.class);
-                final PsiParameter[] parameters = PsiTreeUtil.getParentOfType(targetNode, PsiMethod.class).getParameterList().getParameters();
+        codeActions.add(JDTUtils.createCodeAction(context, diagnostic, title, getParticipantId()));
+    }
 
-                final String TITLE_MESSAGE = Messages.getMessage("RemoveAllEntityParametersExcept",
-                        parameters[entityParamIndex].getName());
+    private static String getTitle(PsiParameter parameters) {
+        return Messages.getMessage("RemoveAllEntityParametersExcept",
+                parameters.getName());
+    }
 
-                final List<PsiParameter> entityParams = new ArrayList<>();
-                entityParamIndexes.forEach(x -> {
-                    if (!x.equals(entityParamIndex)) {
-                        entityParams.add(parameters[x]);
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+        final CodeAction toResolve = context.getUnresolved();
+        final PsiElement node = context.getCoveredNode();
+        final PsiClass parentType = PsiTreeUtil.getParentOfType(node, PsiClass.class);
+        final PsiMethod parentMethod = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
+        assert parentMethod != null;
+        final PsiParameterList parameterList = parentMethod.getParameterList();
+        final List<PsiParameter> entityParams = new ArrayList<>();
+        String title = toResolve.getTitle();
+
+        if (parameterList.getParametersCount() > 0) {
+            final PsiParameter[] parameters = parameterList.getParameters();
+            for (PsiParameter parameter : parameters) {
+                if (isEntityParam(parameter)) {
+                    if (!title.contains(parameter.getName())) {
```

Comment:
contains() isn't going to work in general. What if the parameter name is one of the other words in the title or is contained in the name of another parameter? This was checking indexes before. I think the list should have been pre-computed and saved in the CodeAction. See `io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.codeaction.InsertAnnotationMissingQuickFix` as an example of saving extended data.

Reply:
Modifications done. Please have a look.